src/models/emotion_detection/emotion_labels.py

```
@@ -0,0 +1,346 @@
+        emotion: Emotion name
+        
+    Returns:
+        Index of the emotion (0-27)
+        
+    Raises:
+        ValueError: If emotion is not found
+    """
+    try:
+        return GOEMOTIONS_EMOTIONS.index(emotion.lower())
+    except ValueError:
+        raise ValueError(f"Emotion '{emotion}' not found in GoEmotions list")
```

Ruff (0.12.2)

Within an `except` clause, raise exceptions with `raise ... from err` or `raise ... from None` to distinguish them from errors in exception handling

(B904)